packages/api/test/e2e/promise-queries.spec.ts

```diff
@@ -0,0 +1,196 @@
+// Copyright 2017-2019 @polkadot/api authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import BN from 'bn.js';
+
+import WsProvider from '@polkadot/rpc-provider/ws';
+import testingPairs from '@polkadot/keyring/testingPairs';
+import { LinkageResult } from '@polkadot/types/codec/Linkage';
+import { EventRecord, Header, Vector } from '@polkadot/types';
+
+import Api from './../../src/promise';
+
+const ZERO = new BN(0);
+const WS_URL = 'ws://127.0.0.1:9944';
+// const WS_URL = 'wss://poc3-rpc.polkadot.io/';
+
+describe.skip('Promise e2e queries', () => {
+  const keyring = testingPairs({ type: 'ed25519' });
+  let api: Api;
+
+  beforeEach(async (done) => {
+    if (!api) {
+      api = await Api.create({
+        provider: new WsProvider(WS_URL)
+      });
+    }
+
+    jest.setTimeout(30000);
+    done();
+  });
+
+  it('makes the runtime, rpc, state & extrinsics available', () => {
+    expect(api.genesisHash).toBeDefined();
+    expect(api.runtimeMetadata).toBeDefined();
+    expect(api.runtimeVersion).toBeDefined();
+    expect(api.rpc).toBeDefined();
+    expect(api.query).toBeDefined();
+    expect(api.tx).toBeDefined();
+    expect(api.derive).toBeDefined();
+  });
+
+  it('queries state for a balance', async () => {
+    return(
+      api.query.balances.freeBalance(keyring.alice.address(), balance => {
```

Comment:
Generally you could just do `return api.query.balances.freeBalance(keyring.alice.address(), balance => {` (all tests, but it is neither here, nor there. (So nothing to do)

Reply:
(nit) prefer `(balance) =>` as opposed to the bare version

Reply:
Whoops, merged before reading this. Github did show me an 'approved' from you so there must have been an overlap.
I'll address your comments in another PR

Reply:
https://github.com/polkadot-js/api/pull/955